src/gradient/sensitivity_calculator.py

```diff
@@ -80,9 +97,19 @@ def run(self, scale: str, params: dict):
         # 6. Calculate eigenvalue and eigenvalue gradients for R0
         self._calculate_eigenvectors()
 
+        # 7. Model simulation and getting their resulting final epidemic size after contact change
+        if self.model == "kenya":
+            self.model = EpidemicModelBase(model_data=self.kenyan_aggregated_data)
+        else:
+            self.model = EpidemicModelBase(model_data=self.data)
+        self._choose_model(epi_model=self.epidemic_model)
+        self.susceptibles = self.model.get_initial_values()[self.model.c_idx["s"] *
```

Comment:
You overwrite here `self.model` which is a dumb idea. Create a variable `self.sim_model` and assign the simulation model to that. Additionally, `EpidemicModel` is an abstract class, thus it cannot be instantiated ever - if you want to initialize the variable, then use `None`.